src/subscription_manager/cache.py

```diff
@@ -426,8 +439,35 @@ def has_changed(self):
         return not cached_profile == self.current_profile
 
     def _sync_with_server(self, uep, consumer_uuid, *args, **kwargs):
-        uep.updatePackageProfile(consumer_uuid,
-                self.current_profile.collect())
+        """
+        This method has to be able to sync combined profile, when server supports this functionality
+        and it also has to be able to send only profile containing list of installed RPMs.
+        """
+        combined_profile = self.current_profile
+        if uep.has_capability("combined reporting"):
+            _combined_profile = [
+                {
+                    "content_type": "rpm",
+                    "profile": combined_profile["rpm"]
+                },
+                {
+                    "content_type": "enabled_repos",
+                    "profile": combined_profile["enabled_repos"]
+                },
+                {
+                    "content_type": "modulemd",
+                    "profile": combined_profile["modulemd"]
+                },
+            ]
+            uep.updateCombinedProfile(
+                consumer_uuid,
+                _combined_profile
+            )
```

Comment:
Is it possible to write a couple of basic test cases in `test_cache.py` that exercise this code path (when `uep.has_capability("combined reporting")` is true?)